### DISCUSSION TITLE
Utility freezes after accepting login when trying to download by pubfile id

### DISCUSSION THREAD

**Discussion by bmelikant** [SteamRE/DepotDownloader#228] [Q&A] [answered]
I have been trying to use this utility to build scripts to automatically install mods for Arma 3 on a Linux dedicated server. I built the latest release using dotnet build on Ubuntu 20.04. I wrapped the call to DepotDownloader in a bash script as such:

`./DepotDownloader -username [username] -password [password] -remember-password -dir [path-to-destination] -app 107410 -pubfile 450814997`

The downloader application starts, I get successfully logged in to Steam, but then... the utility just freezes. Sometimes I get a message about accepting a new token, usually it just says "Got session token!" then does nothing else. About once every 20 or so connection attempts result in a successful download of one mod file. As soon as the utility hits the next mod in the list, it freezes again.

The utility correctly creates the output directory for the mod files, so it seems like this may be some network issue or something? It's beyond my knowledge with the Steam API to diagnose it though, so I'm hoping someone can point me in the right direction!

**Comment by yaakov-h** [comment, marked as answer]
possible duplicate of #226. can you please check there?

**Reply by bmelikant** [reply]
The issue appears to me to be related; however, I am using the release specified in the linked issue and am still encountering the issue. I did end up running with -debug, and here is the output from that run:

`
Connecting to Steam3...[ServerList] Resolving server list
[ServerList] Server list provider had no entries, will query SteamDirectory
02:24:33.1534675  System.Net.Http.RequestStart(scheme: https, host: api.steampowered.com, port: 443, pathAndQuery: /ISteamDirectory/GetCMList/v1/?cellid=0&format=vdf, versionMajor: 1, versionMinor: 1, versionPolicy: 0)
02:24:33.1648525  System.Net.NameResolution.ResolutionStart(hostNameOrAddress: api.steampowered.com)
02:24:33.1744286  System.Net.NameResolution.ResolutionStop()
02:24:33.1748009  System.Net.Sockets.ConnectStart(address: InterNetwork:16:{1,187,23,208,45,68,0,0,0,0,0,0,0,0})
02:24:33.1986868  System.Net.Sockets.ConnectStop()
02:24:33.2005665  System.Net.Security.HandshakeStart(isServer: False, targetHost: api.steampowered.com)
02:24:33.2769794  System.Net.Security.HandshakeStop(protocol: 12288)
02:24:33.2783560  System.Net.Http.ConnectionEstablished(versionMajor: 1, versionMinor: 1)
02:24:33.2802443  System.Net.Http.RequestHeadersStart()
02:24:33.2807138  System.Net.Http.RequestHeadersStop()
02:24:33.4103332  System.Net.Http.ResponseHeadersStart()
02:24:33.4116258  System.Net.Http.ResponseHeadersStop()
02:24:33.4155910  System.Net.Http.ResponseContentStart()
02:24:33.4206607  System.Net.Http.ResponseContentStop()
02:24:33.4207422  System.Net.Http.RequestStop()
[ServerList] Resolved 200 servers
[ServerList] Next server candidate: 162.254.193.102:27027 (Tcp)
[141ffade93e34cc0bde025c74f4e9e5e/TcpConnection] Connecting to 162.254.193.102:27027...
02:24:33.4550139  System.Net.Sockets.ConnectStart(address: InterNetwork:16:{105,147,162,254,193,102,0,0,0,0,0,0,0,0})
02:24:33.4836576  System.Net.Sockets.ConnectStop()
[141ffade93e34cc0bde025c74f4e9e5e/TcpConnection] Connected to 162.254.193.102:27027
[141ffade93e34cc0bde025c74f4e9e5e/EnvelopeEncryptedConnection] Got encryption request. Universe: Public Protocol ver: 1
[141ffade93e34cc0bde025c74f4e9e5e/EnvelopeEncryptedConnection] Encryption result: OK
 Done!
Logging '[username]' into Steam3... Done!
Using Steam3 suggested CellID: 80
Got 59 licenses for account!
Got session token!
02:24:34.2374402  System.Net.Http.RequestStart(scheme: https, host: api.steampowered.com, port: 443, pathAndQuery: /IContentServerDirectoryService/GetServersForSteamPipe/v1/?cell_id=80&format=vdf, versionMajor: 1, versionMinor: 1, versionPolicy: 0)
02:24:34.2376685  System.Net.NameResolution.ResolutionStart(hostNameOrAddress: api.steampowered.com)
02:24:34.2464814  System.Net.NameResolution.ResolutionStop()
02:24:34.2465271  System.Net.Sockets.ConnectStart(address: InterNetwork:16:{1,187,23,208,45,68,0,0,0,0,0,0,0,0})
02:24:34.2683146  System.Net.Sockets.ConnectStop()
02:24:34.2683995  System.Net.Security.HandshakeStart(isServer: False, targetHost: api.steampowered.com)
02:24:34.2935754  System.Net.Security.HandshakeStop(protocol: 12288)
02:24:34.2936855  System.Net.Http.ConnectionEstablished(versionMajor: 1, versionMinor: 1)
02:24:34.2937555  System.Net.Http.RequestHeadersStart()
02:24:34.2938054  System.Net.Http.RequestHeadersStop()
02:24:34.4147378  System.Net.Http.ResponseHeadersStart()
02:24:34.4149174  System.Net.Http.ResponseHeadersStop()
02:24:34.4151155  System.Net.Http.ResponseContentStart()
02:24:34.4152872  System.Net.Http.ResponseContentStop()
02:24:34.4153213  System.Net.Http.RequestStop()
02:24:35.4463526  System.Net.Http.RequestStart(scheme: https, host: api.steampowered.com, port: 443, pathAndQuery: /IContentServerDirectoryService/GetServersForSteamPipe/v1/?cell_id=80&format=vdf, versionMajor: 1, versionMinor: 1, versionPolicy: 0)
02:24:35.4468351  System.Net.NameResolution.ResolutionStart(hostNameOrAddress: api.steampowered.com)
02:24:35.4547499  System.Net.NameResolution.ResolutionStop()
02:24:35.4547821  System.Net.Sockets.ConnectStart(address: InterNetwork:16:{1,187,23,208,45,68,0,0,0,0,0,0,0,0})
02:24:35.4759370  System.Net.Sockets.ConnectStop()
02:24:35.4759851  System.Net.Security.HandshakeStart(isServer: False, targetHost: api.steampowered.com)
02:24:35.5002227  System.Net.Security.HandshakeStop(protocol: 12288)
02:24:35.5003126  System.Net.Http.ConnectionEstablished(versionMajor: 1, versionMinor: 1)
02:24:35.5003610  System.Net.Http.RequestHeadersStart()
02:24:35.5003989  System.Net.Http.RequestHeadersStop()
02:24:35.6415839  System.Net.Http.ResponseHeadersStart()
02:24:35.6416206  System.Net.Http.ResponseHeadersStop()
02:24:35.6416522  System.Net.Http.ResponseContentStart()
02:24:35.6416718  System.Net.Http.ResponseContentStop()
02:24:35.6416759  System.Net.Http.RequestStop()
Accepted new login key for account [useraccount]
02:24:35.7228133  System.Net.Http.ConnectionClosed(versionMajor: 1, versionMinor: 1)
02:24:35.7231046  System.Net.Http.ConnectionClosed(versionMajor: 1, versionMinor: 1)
`
I double-checked the SHA for the release I checked out just to make sure everything was right. Should I be grabbing the zip file on the releases page rather than checking out the release tag and building?

**Reply by bmelikant** [reply]
I actually just answered my own question. From the releases page, I downloaded the latest zip file with the hotfix notation and am now seeing the downloads for all the mods in question succeeding. I'm going to mark this question as answered. Thank you!
